app/Http/Controllers/Api/ActivityController.php

```diff
@@ -27,9 +27,22 @@ public function index(Request $request)
     {
         // Create an empty Signup query and eager-load posts, which we
         // can either filter or paginate to retrieve all signup records.
-        $query = $this->newQuery(Signup::class)->with('posts');
+        $query = $this->newQuery(Signup::class);
 
         $filters = $request->query('filter');
+
+        if (array_has($filters, 'status')) {
+            // Constrain Eager Load, only returning posts with status query.
+            $query = $query->with(['posts' => function ($query) use ($filters) {
+                $query->where('status', $filters['status']);
+            }]);
+
```

Comment:
@DFurnes we don't necessarily need this anymore since we are using the new `/posts` endpoint with siblings(!) but should I keep in case we need in the future? cc @sbsmith86